external/CMakeLists.txt

```diff
@@ -35,8 +35,13 @@
 # ...except for FreeBSD, because FreeBSD is a special case that doesn't play well with
 # others.
 
+if(NOT MSVC)
+  add_compile_options(-D_GNU_SOURCE)
```

Comment:
Is this needed since we add `_GNU_SOURCE` to compile options in the root CMakeLists.txt?

Reply:
Yes, since we add it to `CMAKE_C_FLAGS` [1] and it does not pass it to projects added via `add_subdirectory`.  It fixes this build issue [2].


1. https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/CMakeLists.txt#L910C7-L910C20
2. https://github.com/monero-project/monero/issues/9359